src/plugins/spraymon/spraymon.cpp

```diff
@@ -0,0 +1,321 @@
+* By sending these changes, it is understood (unless you specify          *
+* otherwise) that you are offering unlimited, non-exclusive right to      *
+* reuse, modify, and relicense the code.  DRAKVUF will always be          *
+* available Open Source, but this is important because the inability to   *
+* relicense code has caused devastating problems for other Free Software  *
+* projects (such as KDE and NASM).                                        *
+* To specify special license conditions of your contributions, just say   *
+* so when you send them.                                                  *
+*                                                                         *
+* This program is distributed in the hope that it will be useful, but     *
+* WITHOUT ANY WARRANTY; without even the implied warranty of              *
+* MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the DRAKVUF   *
+* license file for more details (it's in a COPYING file included with     *
+* DRAKVUF, and also available from                                        *
+* https://github.com/tklengyel/drakvuf/COPYING)                           *
+*                                                                         *
+***************************************************************************/
+
+#include <libvmi/libvmi.h>
+#include <libdrakvuf/libdrakvuf.h>
+#include <json-c/json_object.h>
+
+#include "spraymon.h"
+#include "plugins/output_format.h"
+
+bool spraymon::read_counter(drakvuf_t drakvuf, addr_t vaddr, vmi_pid_t pid, uint16_t* value)
+{
+    vmi_lock_guard vmi(drakvuf);
+    return (VMI_SUCCESS == vmi_read_16_va(vmi, vaddr, pid, value));
+}
+
+bool spraymon::read_kernel_addr(drakvuf_t drakvuf, addr_t in_address, vmi_pid_t pid, addr_t* out_address)
+{
+    vmi_lock_guard vmi(drakvuf);
+    return (VMI_SUCCESS == vmi_read_addr_va(vmi, in_address, pid, out_address));
+}
+
+bool spraymon::get_counters(drakvuf_t drakvuf, addr_t process, vmi_pid_t pid, uint16_t* gdi_max_count, uint16_t* usr_max_count)
+{
+    addr_t win32process;
+
+    if (!read_kernel_addr(drakvuf, process + this->eprocess_win32process, pid, &win32process))
+    {
+        PRINT_DEBUG("[SPRAYMON] Failed to read EPROCESS->Win32Process\n");
+        return false;
+    }
+
+    if (!win32process)
+    {
+        PRINT_DEBUG("[SPRAYMON] Win32Process is NULL\n");
+        return false;
+    }
+    if (!read_counter(drakvuf, win32process + this->gdihandlecountpeak, pid, gdi_max_count))
+    {
+        PRINT_DEBUG("[SPRAYMON] Failed to read GDI peak handle count\n");
+        return false;
+    }
+
+    if (!read_counter(drakvuf, win32process + this->userhandlecountpeak, pid, usr_max_count))
+    {
+        PRINT_DEBUG("[SPRAYMON] Failed to read USER peak handle count\n");
+        return false;
+    }
+    return true;
+}
+
+void spraymon::compare(drakvuf_t drakvuf, uint16_t gdi_max_count, uint16_t usr_max_count, char* process_name, vmi_pid_t pid)
```

Comment:
misleading name as it doesn't really compare stuff, main reason is to output some log.